chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/FieldLengthReviewer.java

```diff
@@ -1,68 +1,99 @@
-            checkField(listing, listing.getDeveloperAddress().getState(), "developerState");
-            checkField(listing, listing.getDeveloperAddress().getZipcode(), "developerZip");
-        }  else {
-            checkField(listing, listing.getDeveloperStreetAddress(), "developerStreetAddress");
-            checkField(listing, listing.getDeveloperCity(), "developerCity");
-            checkField(listing, listing.getDeveloperState(), "developerState");
-            checkField(listing, listing.getDeveloperZipCode(), "developerZip");
+            checkField(listing, listing.getDeveloperAddress().getStreetLineOne(), "developerStreetAddress", ERROR);
+            checkField(listing, listing.getDeveloperAddress().getStreetLineTwo(), "developerStreetAddressTwo", ERROR);
+            checkField(listing, listing.getDeveloperAddress().getCity(), "developerCity", ERROR);
+            checkField(listing, listing.getDeveloperAddress().getState(), "developerState", ERROR);
+            checkField(listing, listing.getDeveloperAddress().getZipcode(), "developerZip", ERROR);
+        } else {
+            checkField(listing, listing.getDeveloperStreetAddress(), "developerStreetAddress", ERROR);
+            checkField(listing, listing.getDeveloperCity(), "developerCity", ERROR);
+            checkField(listing, listing.getDeveloperState(), "developerState", ERROR);
+            checkField(listing, listing.getDeveloperZipCode(), "developerZip", ERROR);
         }
-        checkField(listing, listing.getDeveloperWebsite(), "developerWebsite");
-        checkField(listing, listing.getDeveloperEmail(), "developerEmail");
-        checkField(listing, listing.getDeveloperPhoneNumber(), "developerPhone");
-        checkField(listing, listing.getDeveloperContactName(), "developerContactName");
+        checkField(listing, listing.getDeveloperWebsite(), "developerWebsite", ERROR);
+        checkField(listing, listing.getDeveloperEmail(), "developerEmail", ERROR);
+        checkField(listing, listing.getDeveloperPhoneNumber(), "developerPhone", ERROR);
+        checkField(listing, listing.getDeveloperContactName(), "developerContactName", ERROR);
 
     }
 
-    private void checkField(final PendingCertifiedProductDTO listing, final Object field, final String errorField) {
-        if (field instanceof Long) {
-            Long fieldCasted = (Long) field;
-            if (fieldCasted.toString().length() > getMaxLength("maxLength." + errorField)) {
-                PendingCertifiedProductDTO productCasted = (PendingCertifiedProductDTO) listing;
-                productCasted.getErrorMessages().add(msgUtil.getMessage("listing." + errorField + ".maxlength",
-                        String.valueOf(getMaxLength("maxLength." + errorField)), fieldCasted));
+    private void checkField(final PendingCertifiedProductDTO listing, final Object field, final String errorField,
+            String type) {
+        if (type.equals(ERROR)) {
```

Comment:
The "if long vs if string" stuff is duplicated. Why not generate the message first, independent of whether it's an ERROR or a WARNING, and only at the end add an:
```
if type.equals(ERROR)
  listing.geterrormessages.add(message)
else
  listing.getwarningmessages.add(message)
```